exporters/ostream/src/log_exporter.cc

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/exporters/ostream/log_exporter.h"
+
+#include <iostream>
+
+namespace nostd   = opentelemetry::nostd;
+namespace sdklogs = opentelemetry::sdk::logs;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+OStreamLogExporter::OStreamLogExporter(std::ostream &sout) noexcept : sout_(sout) {}
+
+std::unique_ptr<sdklogs::Recordable> OStreamLogExporter::MakeRecordable() noexcept
```

Comment:
question: wondering if here it would make sense for `MakeRecordable()` should return a `nullptr` if the exporter is already shutdown?

Reply:
No strong preference, though it will allow us to fail early.

Reply:
Will make the change then if there's no other reason against it :)

Reply:
Hmm on second thought, might be better to leave it as is actually, in case the caller is unaware of the reason `nullptr` is being thrown (e.g. exception or shutdown) -  might be later to be introduced consistently across traces/metrics/logs later - so I've left it unchanged :)